### PR TITLE
doc: add label and notes for experimental features

### PR DESCRIPTION
**Pull request by yizha1**
This PR partly resolves issue #576 
- Lable flag `--signature-manifest` as experimental feature
- Add notes for using artifact manifest

Signed-off-by: Yi Zha <yizha1@microsoft.com>

### CODE REVIEW

File: specs/commandline/sign.md [hunk 1]
```diff
@@ -36,15 +36,15 @@ Flags:
        --plain-http                 registry access via plain HTTP
        --plugin-config stringArray  {key}={value} pairs that are passed as it is to a plugin, refer plugin's documentation to set appropriate values
        --signature-format string    signature envelope format, options: "jws", "cose" (default "jws")
-       --signature-manifest string  manifest type for signature, options: "image", "artifact" (default "image")
+       --signature-manifest string  [Experimental] manifest type for signature, options: "image", "artifact" (default "image")
```
Comment (priteshbandi) [inline specs/commandline/sign.md:39]: We also need to update command help to reflect it as experimental.

We should call out that features/flags mark experimental can be modified and dropped in future. Also, Notation dont guaranty backward comparability from them.

File: specs/commandline/sign.md [hunk 2]
```diff
@@ -154,6 +154,8 @@ Successfully signed localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da
 
 ### Sign an artifact and store the signature using OCI artifact manifest
 
+Note: This is an experimental feature.
+
```
Comment (patrickzheng200) [inline specs/commandline/sign.md:158]:
```suggestion
### [Experimental] Sign an artifact and store the signature using OCI artifact manifest
```

File: specs/commandline/sign.md (hunk 1 above)
Comment (yizha1) [reply specs/commandline/sign.md:39]: @priteshbandi  please check the update in below paragraph.

**Review by patrickzheng200 — APPROVED** [review]
LGTM

**Review by FeynmanZhou — APPROVED** [review]
LGTM

**Review by priteshbandi — APPROVED** [review]
LGTM

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**